### ISSUE TITLE
no-fallthrough treats comments as case statements

### ISSUE THREAD

**Issue by rudym87** [eslint/eslint#5898]
**What version are you using?**
2.8.0 (2.7.0 didn't show this behaviour)

**What did you do?**
My configuration
```javascript
{
    "extends": "eslint:recommended"
}
```
My code
```javascript
var foo;
switch (foo) {
  case 1:
  //case 2:
  case 3:
    break;
  default:
    break;
}
```
**What happened?**
```
5:3 - Expected a 'break' statement before 'case'. (no-fallthrough)
```
**What did you expect to happen?**
No errors reported.
I used the comment to disable a case.
A similar error occurs when I used a comment to group cases like this:
```javascript
var foo;
switch (foo) {
  //group a
  case 1:
  case 2:
  //group b
  case 3:
  case 4:
    break;
  default:
    break;
}
```
In both cases there was no intentional fallthrough without comments, so to me it seem a little bit wrong to fix it with a comment which adheres to `commentPattern`.

**Comment by nzakas** [comment]
This was an intentional change. You can see the discussion here: https://github.com/eslint/eslint/issues/5799

**Comment by alberto** [comment]
Closing since it was by design
